src/queryIterator.ts

```diff
@@ -135,27 +125,29 @@ export class QueryIterator<T> {
   /**
    * Retrieve all the elements of the feed and pass them as an array to a function
    */
-  public async toArray(): Promise<Response<T[]>> {
+
+  public async toArray(): Promise<FeedResponse<T>> {
```

Comment:
Do we need a new special type for the `toArray` repsonse that remembers all the headers? I think the suffers from the same issue we have in v2 where toArray only has the last header values

Reply:
This is a good point. I'll think on it.

Reply:
We're not saving any headers anymore, so 1 FeedResponse is good enough.